### PR TITLE
Improve DateTime class

### PR DESCRIPTION
**Pull request by mikee47**
* Tidy #includes
* Add getMonthDays() to simplify calculation and remove duplicate code
* Add NULL parameter checks in convertFromUnixTime()
* Replace magic numbers with constants already defined in DateTime.h
* Put all constant data into flash
* Simplify month name lookup using CStringArray
* Add member variable initializers & remove from constructor
* Coding style applied
* Use m_snprintf() instead of sprintf() - safer

Note that DateTime does not handle negative values, yet everything uses signed integers. It could be changed, except with convertFromUnixTime() which would break compatibility.

### CODE REVIEW

File: Sming/Services/DateTime/DateTime.h [hunk 1]
```diff
@@ -179,14 +181,14 @@ class DateTime
...
 public:
-	int8_t Hour; ///< Hour (0-23)
-	int8_t Minute; ///< Minute (0-59)
-	int8_t Second; ///< Second (0-59)
-	int16_t Milliseconds; ///< Milliseconds (0-999)
-	int8_t Day; ///< Day of month (1-31)
-	int8_t DayofWeek; ///< Day of week (0-6 Sunday is day 0)
-	int8_t Month; ///< Month (0-11 Jan is month 0)
-	int16_t Year;  ///< Full Year number
+	int8_t Hour = 0; ///< Hour (0-23)
```
[the remark this answers is not in the document]
Comment (mikee47) [reply Sming/Services/DateTime/DateTime.h]: Isn't that a BC?

[the remark this answers is not in the document]
Comment (slaff) [reply Sming/Services/DateTime/DateTime.h]: > Isn't that a BC?

Ou yes.. leave as it is then.

[the remark this answers is not in the document]
Comment (slaff) [reply Sming/Services/DateTime/DateTime.h]: Before I merge it let me ask my usual question: Is that PR tested and working as expected?

[the remark this answers is not in the document]
Comment (mikee47) [reply Sming/Services/DateTime/DateTime.h]: Tested and confirmed:
* parseHttpDate()
* toUnixTime(), convertToUnixTime()
* setTime()

```
5096393 parseHttpDate("Sun, 06 Nov 1994 08:49:37 GMT") produced "06.11.1994 08:49:37"
5096731 unixTime = 784111777 [toUnixTime()]
5096920 "06.11.1994 08:49:37" [setTime()]
```

**Pull request merged**